structuretoolkit/analyse/snap.py

```diff
@@ -0,0 +1,397 @@
+    return np.array(
+        [
+            np.concatenate(
+                (
+                    atom,
+                    convert_mat(
+                        mat=np.dot(
+                            atom.reshape(len(atom), 1), atom.reshape(len(atom), 1).T
+                        )
+                    ),
+                )
+            )
+            for atom in linear_per_atom
+        ]
+    )
+
+
+def calc_sum_quad(linear_sum):
+    return np.concatenate(
+        (
+            linear_sum,
+            convert_mat(
+                mat=np.dot(
+                    linear_sum.reshape(len(linear_sum), 1),
+                    linear_sum.reshape(len(linear_sum), 1).T,
+                )
+            ),
+        )
+    )
+
+
+def get_apre(cell):
+    a, b, c = cell
+    an, bn, cn = [np.linalg.norm(v) for v in cell]
+
+    alpha = np.arccos(np.dot(b, c) / (bn * cn))
+    beta = np.arccos(np.dot(a, c) / (an * cn))
+    gamma = np.arccos(np.dot(a, b) / (an * bn))
+
+    xhi = an
+    xyp = np.cos(gamma) * bn
+    yhi = np.sin(gamma) * bn
+    xzp = np.cos(beta) * cn
+    yzp = (bn * cn * np.cos(alpha) - xyp * xzp) / yhi
+    zhi = np.sqrt(cn**2 - xzp**2 - yzp**2)
+
+    return np.array(((xhi, 0, 0), (xyp, yhi, 0), (xzp, yzp, zhi)))
+
+
+def write_ase_structure(lmp, structure):
+    number_species = len(set(structure.get_chemical_symbols()))
+
+    apre = get_apre(cell=structure.cell)
+    (
+        (xhi, xy, xz),
+        (_, yhi, yz),
+        (_, _, zhi),
+    ) = apre.T
+    lmp.command(
+        "region 1 prism"
+        + " 0.0 "
+        + str(xhi)
+        + " 0.0 "
+        + str(yhi)
+        + " 0.0 "
+        + str(zhi)
+        + " "
+        + str(xy)
+        + " "
+        + str(xz)
+        + " "
+        + str(yz)
+        + " units box"
+    )
+    lmp.command("create_box " + str(number_species) + " 1")
+
+    el_dict = {el: i for i, el in enumerate(set(structure.get_chemical_symbols()))}
+
+    R = np.dot(np.linalg.inv(structure.cell), apre)
+
+    positions = structure.positions.flatten()
+    if np.matrix.trace(R) != 3:
+        positions = np.array(positions).reshape(-1, 3)
+        positions = np.matmul(positions, R)
+    positions = positions.flatten()
+    elem_all = np.array([el_dict[el] + 1 for el in structure.get_chemical_symbols()])
+    lmp.create_atoms(
+        n=len(structure),
+        id=None,
+        type=(len(elem_all) * c_int)(*elem_all),
+        x=(len(positions) * c_double)(*positions),
+        v=None,
+        image=None,
+        shrinkexceed=False,
+    )
+
+
+def extract_compute_np(lmp, name, compute_type, result_type, array_shape):
+    """
+    Convert a lammps compute to a numpy array.
+    Assumes the compute returns a floating point numbers.
+    Note that the result is a view into the original memory.
+    If the result type is 0 (scalar) then conversion to numpy is skipped and a python float is returned.
+    """
+    ptr = lmp.extract_compute(
```

Comment:
Why not just add the extra four characters to call it "pointer"? Since the docstring bothers mentioning about the memory I assume that's the justification for this name

Reply:
Done